File: backend/search-service/src/main/java/com/codecanvas/searchservice/service/search/SearchService.java
package com.codecanvas.searchservice.service.search;

import com.codecanvas.searchservice.dto.request.SearchRequest;
import com.codecanvas.searchservice.dto.response.*;
import com.codecanvas.searchservice.dto.request.IndexSnippetRequest;
import com.codecanvas.searchservice.kafka.event.UserRegisteredEvent;
import com.codecanvas.searchservice.kafka.event.UserUpdatedEvent;
import com.codecanvas.searchservice.kafka.event.UserRegisteredEvent;
import com.codecanvas.searchservice.kafka.event.UserUpdatedEvent;

import java.util.List;
import java.util.UUID;

public interface SearchService {

    SearchPageResponse search(SearchRequest request, UUID userId);

    void saveSearchHistory(UUID userId, SearchRequest request);


    List<SearchHistoryResponse> getUserSearchHistory(UUID userId);

    List<AutocompleteResponse> getSuggestions(String keyword);

    List<PopularSearchResponse> getPopularSearches();

    void indexSnippet(IndexSnippetRequest request);

    void deleteSnippet(UUID snippetId);


    /*
     * =========================================================
     * USER OPERATIONS
     * =========================================================
     */

    void indexUser(UserRegisteredEvent event);

    void updateUser(UserUpdatedEvent event);

    void deleteUser(UUID userId);


    /*
     * =========================================================
     * USER SEARCH
     * Search users by full name, username or bio
     * =========================================================
     */
    List<UserSearchResponse> searchUsers(String keyword);

    /*
     * =========================================================
     * USER AUTOCOMPLETE
     * Suggest users by full name or username
     * =========================================================
     */
    List<UserAutocompleteResponse> autocompleteUsers(String keyword);
}

File: frontend/src/services/searchService.js
// import axiosInstance from "../api/axios";

// /*
//  * ============================================
//  * SEARCH SNIPPETS
//  * ============================================
//  */
// export const searchSnippets = async (request) => {
//     const response = await axiosInstance.post(
//         "/api/search/snippets",
//         request
//     );

//     return response.data;
// };

// /*
//  * ============================================
//  * SEARCH HISTORY
//  * ============================================
//  */
// export const getSearchHistory = async () => {
//     const response = await axiosInstance.get(
//         "/api/search/history"
//     );

//     return response.data;
// };

// /*
//  * ============================================
//  * POPULAR SEARCHES
//  * ============================================
//  */
// export const getPopularSearches = async () => {
//     const response = await axiosInstance.get(
//         "/api/search/popular"
//     );

//     return response.data;
// };

// /*
//  * ============================================
//  * AUTOCOMPLETE
//  * ============================================
//  */
// export const getSuggestions = async (keyword) => {
//     const response = await axiosInstance.get(
//         "/api/search/suggestions",
//         {
//             params: { keyword },
//         }
//     );

//     return response.data;
// };

// /*
//  * ============================================
//  * USER SEARCH
//  * ============================================
//  */
// export const searchUsers = async (keyword) => {
//     const response = await axiosInstance.get(
//         "/api/search/users",
//         {
//             params: { keyword },
//         }
//     );

//     return response.data;
// };




import axiosInstance from "../api/axios";

/*
 * ============================================
 * SEARCH SNIPPETS
 * ============================================
 */
export const searchSnippets = async (request) => {
  const response = await axiosInstance.post(
    "/api/search/snippets",
    request
  );

  return response.data;
};

/*
 * ============================================
 * SEARCH HISTORY
 * ============================================
 */
export const getSearchHistory = async () => {
  const response = await axiosInstance.get(
    "/api/search/history"
  );

  return response.data;
};

/*
 * ============================================
 * POPULAR SEARCHES
 * ============================================
 */
export const getPopularSearches = async () => {
  const response = await axiosInstance.get(
    "/api/search/popular"
  );

  return response.data;
};

/*
 * ============================================
 * SNIPPET AUTOCOMPLETE
 * ============================================
 */
export const getSuggestions = async (keyword) => {
  const response = await axiosInstance.get(
    "/api/search/suggestions",
    {
      params: {
        keyword,
      },
    }
  );

  return response.data;
};

/*
 * ============================================
 * USER AUTOCOMPLETE
 * ============================================
 */
export const getUserSuggestions = async (keyword) => {
    const response = await axiosInstance.get(
        "/api/search/users/suggestions",
        {
            params: { keyword },
        }
    );

     return response.data;
 };